test/oss_test.cpp

```diff
@@ -20,120 +20,172 @@
 #include <unistd.h>	
 #include <gtest/gtest.h>
 
+using namespace emeralddb::oss;
+using namespace std;
+using namespace testing;
+
+
+class ossSocketTest : public Test {
+public:
+  ossSocketTest() {}
+  ~ossSocketTest() {}
+
+protected: 
+  virtual void SetUp() {
+    cout << "before test" << endl; 
+    _port = genRandomPort(); 
+    _timeout = 6; 
+    ossPtr = getInstance(); 
+  }
+
+  virtual void TearDown() {
+    cout << "after test" << endl; 
+    release(ossPtr);
+    ossPtr = nullptr; 
+    ASSERT_EQ(nullptr, ossPtr); 
+  }
+
+  void initSocket() {
+     cout << "test initSocket" << endl; 
+     ASSERT_EQ(EDB_OK, ossPtr->initSocket()); 
+     ASSERT_TRUE(ossPtr->getFd() > 0); 
+     ASSERT_EQ(true, ossPtr->getInitStatus()); 
+
+     unsigned int hostnameLen = sizeof(char) * 64; 
+     char *hostnamePtr = (char*) malloc(sizeof(char) * 64); 
+     int rc = ossPtr->getHostName(hostnamePtr, hostnameLen); 
+
+     ASSERT_EQ(rc, EDB_OK); 
+     ASSERT_TRUE(strlen(hostnamePtr) > 0 && strlen(hostnamePtr) < sizeof(char) * 64);  
+     free(hostnamePtr); 
+     hostnamePtr = nullptr; 
+  }
+
+  void socketServer() {
+    cout << "test oss socket server setup scenario" << endl; 
+    // server side setup needs 2 steps
+    // step1 init socket and check ok 
+    ASSERT_EQ(EDB_OK, ossPtr->initSocket()); 
+
+    // step2 bind && listen, also print hostname and bind listen got bind ret value and check 
+    ASSERT_EQ(EDB_OK, ossPtr->bind_listen());
+    int hostLen = sizeof(char) * 64; 
+    char *hostnamePtr = (char*) malloc(hostLen); 
+    ASSERT_EQ(EDB_OK, (*ossPtr).getHostName(hostnamePtr, hostLen)); 
+    ASSERT_TRUE(hostnamePtr != nullptr && hostnamePtr[0] != '\0');   
+    int _port = (*ossPtr).getLocalPort(); 
+    cout << "setup server hostname [" << hostnamePtr << "] is listening to [" << _port << "]" << endl; 
+  }
+
+    void release(ossSocket *ossPtr) {
+    if (ossPtr != nullptr) {
+      free(ossPtr);
+    }
+  }
+
+  void connect() {
```

Comment:
in this test case only test client side and server side's connection can be established successfully, but still require a communication like exchanging message test case scenario to verify whether communication is ok.

what's more we also need to design some more corner cases like exchanging more date package exchanging 
that the size of the message package is too large to server/client side to handle. 

or set a high threshold connection to the server side and then create multiple connection requests from client side to server side in which the server side handle too many connection requests to handle, until it gots an exception and exit.